openapi_schema_to_json_schema/to_jsonschema.py

```diff
@@ -43,12 +43,33 @@ def convert(schema, options=None):
     if options['cloneSchema']:
         schema = json.loads(json.dumps(schema))
 
-    schema = convertSchema(schema, options)
+    schema = convertDoc(schema, options)
```

Comment:
I'd rather this `convert` function remain unchanged. 

I'd like `convert` to continue to operate only on schemas, and then add an independent function (`convertDoc`) for full document conversion (preferably in its own file).

Reply:
Done that.

Based on your feedback, I've gone with the stupidly simple and in efficient recursion handling for `paths`, so that there is less code specifically about the doc structure.  I dont have a need for high amounts of efficiency.  The one case where the current code will fail is if there is a path route endpoint of `schema` in the doc - possible, but they should really be using `schema/` or `schema.json` instead.

If you prefer explicit finding of schemas in expected parts of the doc only, I can do that.